### PR TITLE
[oracle] Default metric_prefix (DBMON-3622)

### PR DESCRIPTION
**Pull request by nenadnoveljic**
### What does this PR do?

Set default for `metric_prefix` in `custom_queries` to `oracle`.

### Motivation

Compatibility with the old Python integration.

### Describe how to test/QA your changes

Don't set `metric_prefix` for and custom query and check if the metric is being emitted.

### CODE REVIEW

File: cmd/agent/dist/conf.d/oracle-dbm.d/conf.yaml.example [hunk 1]
```diff
@@ -234,7 +234,7 @@ instances:
   ## @param custom_queries - list of mappings - optional
   ## Each query must have 2 fields, and can have a third optional field:
   ##
-  ## 1. metric_prefix - Each metric starts with the chosen prefix.
+  ## 1. metric_prefix - Each metric starts with the chosen prefix. Default is `oracle`.
```
Comment (urseberry) [inline cmd/agent/dist/conf.d/oracle-dbm.d/conf.yaml.example:237]:
```suggestion
  ## 1. metric_prefix - Each metric starts with the chosen prefix. The default is `oracle`.
```

File: releasenotes/notes/oracle-default-metric-prefix-b309821f0f5dddd3.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    [oracle] Set default for ``metric_prefix`` in ``custom_queries`` to ``oracle``.
```
Comment (urseberry) [inline releasenotes/notes/oracle-default-metric-prefix-b309821f0f5dddd3.yaml:11]:
```suggestion
    [oracle] Set the default for ``metric_prefix`` in ``custom_queries`` to ``oracle``.
```

**Review by urseberry — APPROVED** [review]

**Review by lu-zhengda — APPROVED** [review]

**Comment by sebastianfeduniak** [comment]
@nenadnoveljic Hello! I don't know if that's related but at least it touches the piece of code I have issue with. I am using image `agent:7-jmx` image. In the latest version I see change in the `metric_prefix` implementation. 
For example, with the following configuration:
```
      - metric_prefix: online.oracle.slow-locking-queries
        query: |
```
In datadog for the previous version the metric prefix was `oracle.online.oracle.slow_locking_queries`
now it's `online.oracle.slow_locking_queries` (so `oracle` prefix is deleted).

As a consequence, all my dashboards using metrics from custom queries stopped working. 
What should I do to keep the old prefix? Any simple way or I need to change the `metric_prefix` in all my custom queries?

**Comment by nenadnoveljic** [comment]
`metric_prefix` is absolute, i.e. it shouldn't be automatically prepended by `oracle`. Consequently, to keep the old prefix (`oracle.online.oracle.slow_locking_queries`), `oracle.` should be prepended to the current `metric_prefix` in the configuration file. 

To better understand the problem in the old release, could you provide the following:
- the agent release prior to upgrade to `7.53`
- configuration settings in `oracle.d` (don't forget to remove the password.)

**Comment by sebastianfeduniak** [comment]
Thanks for checking @nenadnoveljic !

Previous version was: `v7.52.0`

Configuration:

```
## All options defined here are available to all instances.
#
integration_name: oracle
#
init_config:

## @param global_custom_queries - list of mappings - optional
## See `custom_queries` defined below.
##
## Global custom queries can be applied to all instances using the
## `use_global_custom_queries` setting at the instance level.
#
# global_custom_queries:
#   - query: <QUERY>
#     columns: <COLUMNS>
#     tags: <TAGS>

## @param service - string - optional
## Attach the tag `service:<SERVICE>` to every metric, event, and service check emitted by this integration.
##
## Additionally, this sets the default `service` for every log source.
#
#service: online.oracle.costumQueries
## Every instance is scheduled independent of the others.
#
instances:
  ## @param server - string - required
  ## The IP address or hostname of the Oracle Database Server.
  #
  - server: xxx
    ## @param service_name - string - required
    ## The Oracle Database service name. To view the services available on your server,
    ## run the following query: `SELECT value FROM v$parameter WHERE name='service_names'`
    #
    service_name: xxx
    ## @param user - string - required
    ## The username for the Datadog user account.
    #
    user: xxx
    ## @param password - string - required
    ## The password for the Datadog user account.
    #
    password: xxx
    ## @param jdbc_driver_path - string - optional - default: <DRIVER_PATH>.jar
    ## Set this parameter if you are not using the oracle native client.
    ## You can also add it to your $CLASSPATH instead.
    #
    jdbc_driver_path: /opt/oracle/ojdbc8.jar
    ## @param only_custom_queries - boolean - optional - default: false
    ## Set this parameter to `true` if you want to skip default system, process, and tablespace metrics checks.
    #
    only_custom_queries: false
    ## @param use_global_custom_queries - string - optional - default: true
    ## How `global_custom_queries` should be used for this instance. There are 3 options:
    ##
    ## 1. true - `global_custom_queries` override `custom_queries`.
    ## 2. false - `custom_queries` override `global_custom_queries`.
    ## 3. extend - `global_custom_queries` are used in addition to any `custom_queries`.
    #
    use_global_custom_queries: "%%env_DD_ORACLE_USE_GLOBAL_CUSTOM_QUERIES%%"

    ## @param custom_queries - list of mappings - optional
    ## Each query must have 2 fields, and can have a third optional field:
    ##
    ## 1. query - The SQL to execute. It can be a simple statement or a multi-line script.
    ##            Use the pipe `|` if you require a multi-line script.
    ## 2. columns - The list representing each column, ordered sequentially from left to right.
    ##              The number of columns must equal the number of columns returned in the query.
    ##              There are 2 required pieces of data:
    ##                a. name - The suffix to append to `<INTEGRATION>.` to form
    ##                          the full metric name. If `type` is `tag`, this column is
    ##                          considered a tag and applied to every
    ##                          metric collected by this particular query.
    ##                b. type - The submission method (gauge, monotonic_count, etc.).
    ##                          This can also be set to `tag` to tag each metric in the row
    ##                          with the name and value of the item in this column. You can
    ##                          use the `count` type to perform aggregation for queries that
    ##                          return multiple rows with the same or no tags.
    ##              Columns without a name are ignored. To skip a column, enter:
    ##                - {}
    ## 3. tags (optional) - A list of tags to apply to each metric.
    #
    custom_queries:
      # Slow running locking queries
      - metric_prefix: online.oracle.slow-locking-queries
        query: |
          SELECT
            <my-query>
        columns:
          - name: object-schema
            type: tag
          - name: object-name
            type: tag
          - name: oracle-username
            type: tag
          - name: max-execution-seconds
            type: gauge
          - name: max-locked-mode
            type: tag
          - name: one-of-session-ids
            type: tag
          - name: one-of-sql-ids
            type: tag
          - name: max-remaining-seconds
            type: gauge
      
    ## @param tags - list of strings - optional
    ## A list of tags to attach to every metric and service check emitted by this instance.
    ##
    ## Learn more about tagging at https://docs.datadoghq.com/tagging
    #
    tags:
      - env:%%env_DD_ENV%%
      - domain:online
      - host:%%env_DD_ORACLE_HOST%%

    ## @param service - string - optional
    ## Attach the tag `service:<SERVICE>` to every metric, event, and service check emitted by this integration.
    ##
    ## Overrides any `service` defined in the `init_config` section.
    #
    service: datadog-oracle

    ## @param min_collection_interval - number - optional - default: 15
    ## This changes the collection interval of the check. For more information, see:
    ## https://docs.datadoghq.com/developers/write_agent_check/#collection-interval
    #
    min_collection_interval: 60

    ## @param empty_default_hostname - boolean - optional - default: false
    ## This forces the check to send metrics with no hostname.
    ##
    ## This is useful for cluster-level checks.
    #
    # empty_default_hostname: false

```

**Comment by nenadnoveljic** [comment]
Thanks. I managed to reproduce the behaviour in `7.52` - it's a bug. The behaviour in the release `7.53` is correct. I'm sorry for the inconvenience.

**Comment by sebastianfeduniak** [comment]
It's not only about `7.52`. I am using it for at least 2 years with no changes to the oracle check configuration. So from my perspective it's a breaking change in `7.53` that can impact anyone using oracle custom queries. What do you think @nenadnoveljic ?

**Comment by nenadnoveljic** [comment]
Yes, that's true. On the one hand, it's an inadvertent breaking change that also affects older releases. On the other hand, it implements the correct behaviour. I'm checking internally how to proceed.

**Comment by sebastianfeduniak** [comment]
ok, got it.

**Pull request merged**